### PR TITLE
Add test for cloning a locked provisioning template

### PR DESCRIPTION
**Pull request by shubhamsg199**
We have tests for cloning a template but that is cloning a newly created template hence, adding a test for cloning an already locked provisioning template.

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_provisioningtemplate.py -k test_positive_clone_locked

**Review by Griffin-Sullivan — APPROVED** [review]
Looks good! If you wanted to parametrize this you could also change `clone_setup` to take in a param for `locked` or `custom` and then return the Provisioning Template you need for the test. That way you could run this all in one test. However, since the other test has some OS coverage too, I think it's fine to have two separate tests.

**Comment by jyejare** [comment]
The PRT tests was passing, but I rebase this PR since PR checks were failing with `codecov` dependency which we already fixed.

File: tests/foreman/ui/test_provisioningtemplate.py [hunk 1]
```diff
@@ -80,6 +80,35 @@ def test_positive_clone(session, clone_setup):
         assert set(clone_setup['os_list']) == {f'{os.name} {os.major}' for os in assigned_oses}
 
 
+@pytest.mark.tier2
+def test_positive_clone_locked(session, target_sat):
```
Comment (jyejare) [inline tests/foreman/ui/test_provisioningtemplate.py:84]: Do you think the `caseLevel` and the `caseImportance` for this test case is same as defined at module level? If not please change is accordingly.

At least I doubt on caseLevel to be acceptance here, I think it should be `Integration` or something else like `system` etc.

File: tests/foreman/ui/test_provisioningtemplate.py [hunk 2]
```diff
@@ -80,6 +80,35 @@ def test_positive_clone(session, clone_setup):
...
+        session.provisioningtemplate.clone(
+            'Kickstart default',
+            {
+                'template.name': clone_name,
+            },
+        )
+        pt = target_sat.api.ProvisioningTemplate().search(query={'search': f'name=={clone_name}'})
+        assert (
+            pt
+        ), 'Template {} expected to exist but is not included in the search' 'results'.format(
```
Comment (jyejare) [inline tests/foreman/ui/test_provisioningtemplate.py:107]: Use `f-strings`

**Review by jyejare — COMMENTED** [review]
ACK pending minor comments!

File: tests/foreman/ui/test_provisioningtemplate.py (hunk 1 above)
Comment (shubhamsg199) [reply tests/foreman/ui/test_provisioningtemplate.py:84]: yes, updated

File: tests/foreman/ui/test_provisioningtemplate.py (hunk 2 above)
Comment (shubhamsg199) [reply tests/foreman/ui/test_provisioningtemplate.py]: updated

**Review by jyejare — APPROVED** [review]

**Pull request merged**